source/core/GenericFunctions.cpp

```diff
@@ -531,6 +544,15 @@ InstructionCore* EmitGenericUnOpInstruction(ClumpParseState* pInstructionBuilder
 
     InstructionCore* pInstruction = nullptr;
     switch (destType->BitEncoding()) {
+        case kEncoding_Variant:
+        {
+            ConstCStr unaryOpName = "DataToVariant";
+            SubString unaryOpToken(unaryOpName);
+            pInstructionBuilder->InternalAddArgFront(nullptr, pInstructionBuilder->_argTypes[0]);
```

Comment:
ah, good catch. Missed to check if it is "Convert" before emitting "DataToVariant". I think we got lucky because other unary operations throw SA error and we may never hit this except "Convert". Fixing it now to add the check. Thanks.

Reply:
You build the arguments the way the instruction expects. For conversion, DataToVariant expects arguments in the order and we have to add in the front. The existing instructions did not run into this situation and they always required additional arguments at the end.